deployments/helm-chart/templates/controller-daemonset.yaml

```diff
@@ -136,14 +123,12 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.name
-{{- if .Values.controller.env }}
-{{ toYaml .Values.controller.env | indent 8 }}
-{{- end }}
-{{- if .Values.nginxServiceMesh.enable }}
         - name: POD_SERVICEACCOUNT
           valueFrom:
             fieldRef:
               fieldPath: spec.serviceAccountName
```

Comment:
This is actually something that we missed in our injection logic. This value is used by us for the SPIFFE certs. I don't think KIC uses it anywhere else.

Reply:
Probably doesn't hurt to have it as part of the official manifest (like you've done here), since it's just an env var. But we likely need to update our controller to inject this if it's not present.

Reply:
@sjberman, I created internal issue NSM-3677 to track this issue.

Reply:
I wonder if we just keep this var gated behind the NSM condition, since only we use it.

Reply:
Reverted this change as suggested.